src/cls/rbd/cls_rbd_types.h

```diff
@@ -328,10 +328,14 @@ struct TrashSnapshotNamespace {
     SNAPSHOT_NAMESPACE_TYPE_TRASH;
 
   std::string original_name;
+  SnapshotNamespaceType original_snapshot_namespace_type =
+    SNAPSHOT_NAMESPACE_TYPE_USER;
```

Comment:
super nit: why not have `original_snapshot_namespace_type` field before `original_name`?

Reply:
... eh, only because if it ever expands to the full namespace, it would need to be encoded at the end.